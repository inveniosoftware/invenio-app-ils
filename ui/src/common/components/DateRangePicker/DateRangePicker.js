import React, { Component } from 'react';
import { Icon, Segment } from 'semantic-ui-react';
import { DateInput } from 'semantic-ui-calendar-react';
import PropTypes from 'prop-types';

export class DateRangePicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      fromDate: props.defaultStart ? props.defaultStart : '',
      toDate: props.defaultEnd ? props.defaultEnd : '',
    };
  }

  handleDateChange = (event, { name, value }) => {
    this.setState({ [name]: value });
    if (this.props.handleDateChange) {
      this.props.handleDateChange(name, value);
    }
  };

  render() {
    return (
      <>
        <Segment>
          <DateInput
            autoComplete="off"
            clearable
            clearIcon={<Icon name="remove" color="red" />}
            closable
            dateFormat="YYYY-MM-DD"
            iconPosition="left"
            maxDate={this.state.toDate}
            name="fromDate"
            onChange={this.handleDateChange}
            placeholder="Start Date"
            value={this.state.fromDate}
            data-test={this.state.fromDate}
          />
        </Segment>
        <Segment>
          <DateInput
            autoComplete="off"
            clearable
            clearIcon={<Icon name="remove" color="red" />}
            closable
            dateFormat="YYYY-MM-DD"
            iconPosition="left"
            minDate={this.state.fromDate}
            name="toDate"
            onChange={this.handleDateChange}
            placeholder="End Date"
            value={this.state.toDate}
            data-test={this.state.toDate}
          />
        </Segment>
      </>
    );
  }
}

DateRangePicker.propTypes = {
  handleDateChange: PropTypes.func,
  defaultStart: PropTypes.string,
  defaultEnd: PropTypes.string,
};
