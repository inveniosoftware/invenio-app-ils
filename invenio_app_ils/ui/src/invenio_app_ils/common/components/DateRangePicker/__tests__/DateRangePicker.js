import React from 'react';
import { mount } from 'enzyme';
import { DateRangePicker } from '../DateRangePicker';

describe('DocumentMetadata tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const mockedChangeDates = jest.fn();

  it('should render Date Range Picker', () => {
    component = mount(
      <DateRangePicker
        handleDateChange={mockedChangeDates}
        defaultStart={'2019-09-03'}
        defaultEnd={'2019-10-03'}
      />
    );
    expect(component).toMatchSnapshot();

    const date = component
      .find('DateRangePicker')
      .find('DateInput')
      .filterWhere(element => element.prop('data-test') === '2019-09-03');
    expect(date).toHaveLength(1);
  });
});
