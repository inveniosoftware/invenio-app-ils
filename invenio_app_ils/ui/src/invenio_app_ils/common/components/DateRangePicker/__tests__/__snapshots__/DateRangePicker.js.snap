// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentMetadata tests should render Date Range Picker 1`] = `
<DateRangePicker
  defaultEnd="2019-10-03"
  defaultStart="2019-09-03"
  handleDateChange={[MockFunction]}
>
  <Segment>
    <div
      className="ui segment"
    >
      <DateInput
        autoComplete="off"
        clearIcon={
          <Icon
            as="i"
            color="red"
            name="remove"
          />
        }
        clearable={true}
        closable={true}
        data-test="2019-09-03"
        dateFormat="YYYY-MM-DD"
        icon="calendar"
        iconPosition="left"
        inline={false}
        localization="en"
        maxDate="2019-10-03"
        name="fromDate"
        onChange={[Function]}
        placeholder="Start Date"
        preserveViewMode={true}
        startMode="day"
        value="2019-09-03"
      >
        <InputView
          animation="scale"
          autoComplete="off"
          clearIcon={
            <Icon
              as="i"
              color="red"
              name="remove"
            />
          }
          clearable={true}
          closeOnMouseLeave={true}
          closePopup={[Function]}
          data-test="2019-09-03"
          duration={200}
          icon="calendar"
          iconPosition="left"
          inline={false}
          name="fromDate"
          onChange={[Function]}
          onFocus={[Function]}
          onMount={[Function]}
          openPopup={[Function]}
          placeholder="Start Date"
          popupIsClosed={true}
          renderPicker={[Function]}
          tabIndex="0"
          value="2019-09-03"
        >
          <FormInputWithRef
            autoComplete="off"
            clearIcon={
              <Icon
                as="i"
                color="red"
                name="remove"
              />
            }
            clearable={true}
            data-test="2019-09-03"
            icon="calendar"
            iconPosition="left"
            innerRef={[Function]}
            name="fromDate"
            onBlur={[Function]}
            onChange={[Function]}
            onClear={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            placeholder="Start Date"
            tabIndex="0"
            value="2019-09-03"
          >
            <FormInput
              as={[Function]}
              autoComplete="off"
              control={[Function]}
              data-test="2019-09-03"
              icon={true}
              iconPosition="left"
              name="fromDate"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              placeholder="Start Date"
              tabIndex="0"
            >
              <FormField
                autoComplete="off"
                control={[Function]}
                data-test="2019-09-03"
                icon={true}
                iconPosition="left"
                name="fromDate"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                placeholder="Start Date"
                tabIndex="0"
              >
                <div
                  className="field"
                >
                  <Input
                    autoComplete="off"
                    data-test="2019-09-03"
                    icon={true}
                    iconPosition="left"
                    name="fromDate"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    placeholder="Start Date"
                    tabIndex="0"
                    type="text"
                  >
                    <div
                      className="ui left icon input"
                      data-test="2019-09-03"
                    >
                      <Icon
                        as="i"
                        color="red"
                        key=".0"
                        link={true}
                        name="remove"
                        onClick={[Function]}
                      >
                        <i
                          aria-hidden="true"
                          className="red remove link icon"
                          onClick={[Function]}
                        />
                      </Icon>
                      <input
                        autoComplete="off"
                        key=".1"
                        name="fromDate"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        placeholder="Start Date"
                        tabIndex="0"
                        type="text"
                        value="2019-09-03"
                      />
                    </div>
                  </Input>
                </div>
              </FormField>
            </FormInput>
          </FormInputWithRef>
          <Transition
            animation="scale"
            duration={200}
            mountOnShow={true}
            onComplete={[Function]}
            transitionOnMount={false}
            unmountOnHide={true}
            visible={false}
          />
        </InputView>
      </DateInput>
    </div>
  </Segment>
  <Segment>
    <div
      className="ui segment"
    >
      <DateInput
        autoComplete="off"
        clearIcon={
          <Icon
            as="i"
            color="red"
            name="remove"
          />
        }
        clearable={true}
        closable={true}
        data-test="2019-10-03"
        dateFormat="YYYY-MM-DD"
        icon="calendar"
        iconPosition="left"
        inline={false}
        localization="en"
        minDate="2019-09-03"
        name="toDate"
        onChange={[Function]}
        placeholder="End Date"
        preserveViewMode={true}
        startMode="day"
        value="2019-10-03"
      >
        <InputView
          animation="scale"
          autoComplete="off"
          clearIcon={
            <Icon
              as="i"
              color="red"
              name="remove"
            />
          }
          clearable={true}
          closeOnMouseLeave={true}
          closePopup={[Function]}
          data-test="2019-10-03"
          duration={200}
          icon="calendar"
          iconPosition="left"
          inline={false}
          name="toDate"
          onChange={[Function]}
          onFocus={[Function]}
          onMount={[Function]}
          openPopup={[Function]}
          placeholder="End Date"
          popupIsClosed={true}
          renderPicker={[Function]}
          tabIndex="0"
          value="2019-10-03"
        >
          <FormInputWithRef
            autoComplete="off"
            clearIcon={
              <Icon
                as="i"
                color="red"
                name="remove"
              />
            }
            clearable={true}
            data-test="2019-10-03"
            icon="calendar"
            iconPosition="left"
            innerRef={[Function]}
            name="toDate"
            onBlur={[Function]}
            onChange={[Function]}
            onClear={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            placeholder="End Date"
            tabIndex="0"
            value="2019-10-03"
          >
            <FormInput
              as={[Function]}
              autoComplete="off"
              control={[Function]}
              data-test="2019-10-03"
              icon={true}
              iconPosition="left"
              name="toDate"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              placeholder="End Date"
              tabIndex="0"
            >
              <FormField
                autoComplete="off"
                control={[Function]}
                data-test="2019-10-03"
                icon={true}
                iconPosition="left"
                name="toDate"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                placeholder="End Date"
                tabIndex="0"
              >
                <div
                  className="field"
                >
                  <Input
                    autoComplete="off"
                    data-test="2019-10-03"
                    icon={true}
                    iconPosition="left"
                    name="toDate"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    placeholder="End Date"
                    tabIndex="0"
                    type="text"
                  >
                    <div
                      className="ui left icon input"
                      data-test="2019-10-03"
                    >
                      <Icon
                        as="i"
                        color="red"
                        key=".0"
                        link={true}
                        name="remove"
                        onClick={[Function]}
                      >
                        <i
                          aria-hidden="true"
                          className="red remove link icon"
                          onClick={[Function]}
                        />
                      </Icon>
                      <input
                        autoComplete="off"
                        key=".1"
                        name="toDate"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        placeholder="End Date"
                        tabIndex="0"
                        type="text"
                        value="2019-10-03"
                      />
                    </div>
                  </Input>
                </div>
              </FormField>
            </FormInput>
          </FormInputWithRef>
          <Transition
            animation="scale"
            duration={200}
            mountOnShow={true}
            onComplete={[Function]}
            transitionOnMount={false}
            unmountOnHide={true}
            visible={false}
          />
        </InputView>
      </DateInput>
    </div>
  </Segment>
</DateRangePicker>
`;
